cloudsync/cs.py

```diff
@@ -260,24 +260,43 @@ def start(self, *, daemon=True, **kwargs):
         self.nmgr.notify(Notification(SourceEnum.SYNC, NotificationType.STARTED, None))
         self.nmgr.start(**kwargs)
 
-
     def stop(self, forever=True, wait=True):
         """
         Stops the cloudsync service.
 
         Args:
             forever: If false is passed, then handles are left open for a future start.  Generally used for tests only.
+            wait: If false, notification manager thread is signaled to stop, but the caller does not wait for it.
+
+        NOTE: The wait parameter does not affect the sync manager or event manager threads. These threads need to write
+        out state on CloudSync shutdown, and as such are always joined.
```

Comment:
I'm not sure this comment is clear -- "as such are always joined" -- does that mean "join always happens", or does it mean "the user of the library is responsible for making super sure that they join these threads, and we're not going to remind you or bug you if you get it wrong, and if you do get it wrong you're introducing the potential for difficult to diagnose bugs"?

Reply:
removed